### PR TITLE
ABC Tests

### LINKED ISSUE #10828: TST: add tests for ABC Types
e.g. something in ``test_common.py`` like

```
p = pd.Period(....)
self.assertTrue(isinstance(p, ABCPeriod))
```

for all of the ABC types defined.

### PR DESCRIPTION
**Pull request by DanNi0130**
closes #10828 

I created tests for this issue here: https://github.com/pydata/pandas/issues/10828. This is my first time working on Pandas, so please let me know if there is anything wrong with the way I have coded these tests. I haven't updated the documentation, I was unsure of what version I should update and whether I should update at all (it seems that the documentation is for additional features and bug fixes rather than increasing test coverage). I'd love any guidance as to what I can do to fully resolve this issue and how I can best help in the future!

### CODE REVIEW

File: pandas/tests/test_common.py [hunk 1]
```diff
@@ -229,8 +271,6 @@ def test_array_equivalent():
     assert not array_equivalent(np.array([np.nan, 1, np.nan]),
                                 np.array([np.nan, 2, np.nan]))
     assert not array_equivalent(np.array(['a', 'b', 'c', 'd']), np.array(['e', 'e']))
-    assert array_equivalent(Float64Index([0, np.nan]), Float64Index([0, np.nan]))
-    assert not array_equivalent(Float64Index([0, np.nan]), Float64Index([1, np.nan]))
```
Comment (jreback) [inline pandas/tests/test_common.py]: is their a reason you removed these?

File: pandas/tests/test_common.py [hunk 2]
```diff
@@ -61,6 +64,45 @@ def __call__(self):
...
+    categorical = pd.Categorical([1, 2, 3, 4], categories=[4, 2, 3, 1])
+    categorical_df = pd.DataFrame({"values": values}, index=categorical)
+    categorical_index = categorical_df.index
+    series = pd.Series(values)
+    df = pd.DataFrame({"names": names}, index=multi_index)
+    panel = df.to_panel()
+    sparse_series = series.to_sparse()
+    sparse_array = pd.SparseArray(np.random.randn(10))
+    period = pd.Period('2012', freq='A-DEC')
+    assert(isinstance(index, com.ABCIndex))
```
Comment (jreback) [inline pandas/tests/test_common.py]: use ``self.assertIsInstance(obj, typ)``

File: pandas/tests/test_common.py [hunk 3]
```diff
@@ -61,6 +64,45 @@ def __call__(self):
...
+    assert(isinstance(int64_index, com.ABCInt64Index))
+    assert(isinstance(float64_index, com.ABCFloat64Index))
+    assert(isinstance(multi_index, com.ABCMultiIndex))
+    assert(isinstance(datetime_index, com.ABCDatetimeIndex))
```
Comment (jreback) [inline pandas/tests/test_common.py]: create these as many of these inline as possible
e.g.

```self.assertIsInstance(pd.Period('2012',freq='M'),com.ABCPeriod)```

File: pandas/tests/test_common.py (hunk 1 above)
Comment (DanNi0130) [inline pandas/tests/test_common.py]: Oh sorry, that was an accident! I will add them back in. Does everything else look okay?

**Pull request closed without merging**